structures/segment-tree/segment-tree.tex

```diff
@@ -0,0 +1,91 @@
+But any updates to the source dataset will require $O(n)$ time to update the prefix sum calculations.
+Frequent changes will quickly show the pitfall of this approach.
+A segment tree is a data structure for storing information about intervals that can be constructed in $O(n \log(n))$ time.
+Whenever the source data changes, update times stay low at $O(\log n)$ time.
+
+% #endif
+
+A segment tree is represented as an array of a size that is dependent on the dataset it is sourced from.
+For a given array of size $n$, a segment tree constructed from it will use up to $2^{\lceil \log_2 (n)\rceil + 1} - 1$ space.
+Once built, the structure of a segment tree cannot changed.
+
+% #ifdef hackpackpp
+
+Generally, a segment tree supports two operations: update and query.
+The update operation will account for changes to values in the source dataset.
+It is even possible to implement a range update operation to change an entire range of values.
+The query operation allows for fast retrieval about segment information.
+In the case of the prefix sum, the update operation will modify the tree's nodes such that it will contain correct information after changes, and the the query operation can return a sum for values in an arbitrary range.
+
+\subsection{Applications}
+
+\begin{itemize}
+    \item range sums in a frequently changing dataset
+    \item range minimum/maximum queries
+\end{itemize}
+
+\subsection{Example Contest Problem: Coming and Going}
+
+In consideration of expanding the cows' barn, Farmer John decided to see just how many of his cows spent any given time of day in the barn.
+To do so, he installed motion sensors on each entryway/exit to the building.
+These sensors are able to detect both the entry and exit of a warm-blooded being.
+This, coupled with his cows' tags, allows him to monitor the movements of specific cows.
+
+The night after the full day in which the sensors were installed, Farmer John sat down to do what he thought was some straightforward analysis, but quickly realized that he was in over his head with only mental math.
+
+Help him figure out which hours each day (he defines as 6AM to 8PM) the barn is the most popular place to be by giving him headcounts as he (slowly) tabulates the data.
+
+\subsubsection{Input Format}
+
+\begin{itemize}
+    \item Line 1: A single integer, $C$, representing the number of cows Farmer John has data on.
+
+    \item The following lines are placed in $C$ groups detailed as follows:
+    \begin{itemize}
+        \item A single line containing two integers, $t_1$ and $t_2$ representing a time range.
+        Should both of these values be 0, Farmer John would not like a count of cows present at any time to be printed.
+        Otherwise, these values are guaranteed to fall between 6 and 20 (inclusive), representing the times between and including 6AM to 8PM.
+        In this case, the highest number of cows seen in the barn at this time should be printed as given in the sample output.
```

Comment:
I didn't say that the output format section was missing. I was saying that information regarding the output format should be in the output fomat section.